cfgov/unprocessed/js/modules/util/tree-traversal.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+/**
+ * Perform a backtrack up a Tree to the root.
+ * Given this Tree and starting at (C):
+ *
+ *        R
+ *       / \
+ *      A   B
+ *    / | \
+ *   C  D  E
+ *
+ * Returns (C) -> (A) -> (R).
+ *
+ * @param {TreeNode} node - Node within a tree to traverse from.
+ * @param {Function} callback - Function to call at each node.
+ */
+function backtrack( node, callback ) {
+  callback.call( this, node );
+  var parent = node.parent;
+  if ( parent ) {
+    backtrack( parent, callback );
+  }
+}
+
+/**
+ * Perform an iterative breadth-first search of a Tree.
+ * Given this Tree and starting at (R):
+ *
+ *        R
+ *       / \
+ *      A   B
+ *    / | \
+ *   C  D  E
+ *
+ * Returns (R) -> (A) -> (B) -> (C) -> (D) -> (E).
+ *
+ * @param {TreeNode} node - Node within a tree to traverse from.
+ * @param {Function} callback - Function to call at each node.
+ */
+function bfs( node, callback ) {
+  var queue = [ node ];
```

Comment:
No, it gets loaded with additional nodes while traversing the tree, see L:49 https://github.com/cfpb/cfgov-refresh/pull/1542/files#diff-327a6706d58da50fa37f20076162e405R49